python/deltalake/table.py

```diff
@@ -412,35 +521,48 @@ def to_pyarrow_table(
         partitions: Optional[List[Tuple[str, str, Any]]] = None,
         columns: Optional[List[str]] = None,
         filesystem: Optional[Union[str, pa_fs.FileSystem]] = None,
+        filters: Optional[
+            Union[List[Tuple[str, str, Any]], List[List[Tuple[str, str, Any]]]]
+        ] = None,
     ) -> pyarrow.Table:
         """
         Build a PyArrow Table using data from the DeltaTable.
 
         :param partitions: A list of partition filters, see help(DeltaTable.files_by_partitions) for filter syntax
         :param columns: The columns to project. This can be a list of column names to include (order and duplicates will be preserved)
         :param filesystem: A concrete implementation of the Pyarrow FileSystem or a fsspec-compatible interface. If None, the first file path will be used to determine the right FileSystem
+        :param filters: A disjunctive normal form (DNF) for filter pushdown. The filters are expressed in a format that is compatible with `pyarrow.compute.Expression`
```

Comment:
I don't think we need to mention PyArrow or pushdown here, as it's an implementation detail, right? But we should be clear that unlike `partitions` it filters rows.

```suggestion
        :param filters: A disjunctive normal form (DNF) predicate for filtering rows. If you pass a filter you do not need to pass ``partitions``.
```

Reply:
> Ideally it should include a case for a conjunctionm one with a disjunction, and one single predicate.
Sure thing! I will add some unit tests.

> I don't think we need to mention PyArrow or pushdown here, as it's an implementation detail, right? But we should be clear that unlike partitions it filters rows.
Agreed. I will fix this.